minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java

```diff
@@ -616,15 +630,21 @@ private static void addConnection(final Element parentElement, Map<String, Objec
             addTextElement(element, "labelIndex", "1");
             addTextElement(element, "zIndex", "0");
 
-            addTextElement(element, "sourceId", "Processor");
+            addTextElement(element, "sourceId", getValueString(connectionProperties, SOURCE_NAME_KEY));
```

Comment:
For properties like these that are required, we should have an indication that it is needed and an associated exception if it is not present.